board: hsdk: add 2 cores configuration for test

vonhust opened the pull request:
add 2 cores configuration for:

 * who want to use 2 cores

 * sanitycheck tests, as we found there are difference between 2 cores and 4 cores, see
      report in issue #26794

Commit:
boards: hsdk: let mdb runner connect cores according to configuration

let mdd runner connect connect cores according to CONFIG_MP_NUM_CPUS,
e.g.
   * CONFIG_MP_NUM_CPUS = 2, just connect 2 cores
   * CONFIG_MP_NUM_CPUS = 1, just connect 1 core
   * CONFIG_MP_NUM_CPUS = 4, connect all 4 cores

Signed-off-by: Wayne Ren <wei.ren@synopsys.com>

ruuddw commented:
Do we really need a new board definition for this? Can't it be handled as a config parameter for the hsdk board in specific tests? (the cmake update does make sense, and the config parameter is already there: why another dts file?)

abrodkin commented:
> Do we really need a new board definition for this? Can't it be handled as a config parameter for the hsdk board in specific tests? (the cmake update does make sense, and the config parameter is already there: why another dts file?)

It's not that we really badly need to have "yet another board" but that's more to simplify tracking of SMP support in Zephyr.
HSDK is the only platform now with quad-core SMP and that 4-core configuration seem to work much worse if compared to more common dual-core setups like for x86_64 for example. It that sense I'm not sure how practical is that to enable N cores for a certain test as I'd ike to have all tests to  be run on dual- & quad-core configs so that we may compare results and see tests which fail only on quad-core configuration.

abrodkin reviewed (changes requested):
We need either an additional openocd configuration for this setup (only 2 cores in of 4 to be used in SMP cluster, like `boards/arc/hsdk/support/openocd-2-cores.cfg`) or similarly to `mdb` factor-in `CONFIG_MP_NUM_CPUS` into existing `boards/arc/hsdk/support/openocd.cfg`.

abrodkin commented on boards/arc/hsdk/hsdk_2cores.dts:
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2019, Synopsys, Inc. All rights reserved.
```
2020

abrodkin replied:
Also there seem to be no difference between 4- & 2-core versions of `.dts` thus it makes sense to move all in `hsdk.dtsi` and include it in both 4- & 2-core versions.

abrodkin reviewed (changes requested):
Would be interesting to make sure if we may have number of cores just as a variable and depending on `CONFIG_MP_NUM_CPUS` value set all the rest. I.e. I don't like duplication of all that for nothing. Though I'm not sure if and how that could be done, sorry.

galak commented:
@ruuddw / @abrodkin  any plans to update this before merge window closes?

ruuddw commented:
I approved earlier already. @abrodkin is right that it can be improved, but main purpose is to increase test coverage for 2 cpu SMP next to 4 cores. I am ok with merging as is.

abrodkin commented:
@IRISZZW could you please look at this one sometime soon (hopefully this Friday so it might be merged in improved state)?

IRISZZW committed:
board: hsdk: add 2 cores configuration for test

add 2 cores configuration for
* who want to use 2 cores
* sanitycheck tests, as we found there are
  difference between 2 cores and 4 cores, see
  report in issue #26794

Signed-off-by: Wayne Ren <wei.ren@synopsys.com>
Signed-off-by: Watson Zeng <zhiwei@synopsys.com>
